### PR TITLE
Profile: Low memory usable

### PR DESCRIPTION
**Pull request by soniasingla**
Signed-off-by: soniasingla <soniasingla.1812@gmail.com>

In this PR, we intend to generate 30d blocks, and in order to avoid overlapping, the `mint` has been increased by 1, `mint := maxt - durToMilis(r)` 🎉

CC: @bwplotka :)

### CODE REVIEW

**Comment by bwplotka** [comment]
Can we name this commit with full sentence? `Profile: Low memory usable`? Usually it's worth to do `Added XYZ` (:

File: pkg/blockgen/profiles.go [hunk 1]
```diff
@@ -26,6 +26,18 @@ func (p ProfileMap) Keys() (keys []string) {
 
 var (
 	Profiles = ProfileMap{
+		"realistic-key-k8s-1d-small": realisticK8s([]time.Duration{
```
Comment (bwplotka) [inline pkg/blockgen/profiles.go:29]: Hm, I think for our case it would be awesome to generate like... 30d (: with just 2-3 series. WDYT? (:

Comment (bwplotka) [reply pkg/blockgen/profiles.go:29]: We can create big blocks (e.g 2w) no problem

and name it `key-k8s-30s-tiny` with `}, 1*time.Hour, 1, 5),` (:

**Review by bwplotka — COMMENTED** [review]
Proposed "long term" version (: 

Maybe even bring it to months if we can use it?

File: pkg/blockgen/profiles.go [hunk 2]
```diff
@@ -26,6 +26,18 @@ func (p ProfileMap) Keys() (keys []string) {
 
 var (
 	Profiles = ProfileMap{
+		"key-k8s-30s-tiny": realisticK8s([]time.Duration{
```
Comment (bwplotka) [inline pkg/blockgen/profiles.go:29]: 30s?

I think we wanted month? :thinking:

File: pkg/blockgen/profiles.go [hunk 3]
```diff
@@ -26,6 +26,18 @@ func (p ProfileMap) Keys() (keys []string) {
...
+			// Two days, from newest to oldest, in the same way Thanos compactor would do.
```
Comment (bwplotka) [inline pkg/blockgen/profiles.go:30]: old comment?

File: pkg/blockgen/profiles.go [hunk 4]
```diff
@@ -26,6 +26,18 @@ func (p ProfileMap) Keys() (keys []string) {
...
+			2 * time.Hour,
+			2 * time.Hour,
+			2 * time.Hour,
+			8 * time.Hour,
+			8 * time.Hour,
+			8 * time.Hour,
+			8 * time.Hour,
```
Comment (bwplotka) [inline pkg/blockgen/profiles.go:37]: Is this really 30d?

File: pkg/blockgen/profiles.go [hunk 5]
```diff
@@ -66,6 +66,19 @@ var (
 			2 * time.Hour,
 			// 10,000 series per block.
 		}, 100, 100),
+
+		"key-k8s-30d-tiny": realisticK8s([]time.Duration{
+			// 30 days, from newest to oldest, in the same way Thanos compactor would do.
```
Comment (bwplotka) [inline pkg/blockgen/profiles.go:71]:
```suggestion
			// 30 days, from newest to oldest.
```

**Review by bwplotka — APPROVED** [review]
One change and LGTM (:

**Comment by soniasingla** [comment]
@bwplotka thanks, updated ❤️

**Review by bwplotka — APPROVED** [review]

**Comment by bwplotka** [comment]
Before we merge, can we add to the PR description why we change dur +1 ?

**Comment by soniasingla** [comment]
Added 👍🏻

**Pull request merged**